src/app/api/routes/alerts.py

```diff
@@ -93,11 +94,20 @@ async def get_alert(alert_id: int = Path(..., gt=0), _=Security(get_current_acce
 
 
 @router.get("/", response_model=List[AlertOut], summary="Get the list of all alerts")
-async def fetch_alerts(_=Security(get_current_access, scopes=[AccessType.admin])):
+async def fetch_alerts(requester=Security(get_current_access, scopes=[AccessType.admin, AccessType.user]),
+                       session=Depends(get_session)):
     """
     Retrieves the list of all alerts and their information
     """
-    return await crud.fetch_all(alerts)
+    if await is_admin_access(requester.id):
+        return await crud.fetch_all(alerts)
+    else:
+        retrieved_alerts = (session.query(models.Alerts)
+                            .join(models.Devices)
+                            .join(models.Accesses)
+                            .filter(models.Accesses.group_id == requester.group_id).all())
```

Comment:
SQL wise, it would be faster & ore efficient to :
- fetch access_ids corresponding to an access in the same group
- fetch device_ids whose access_id is in the previous list
- finally fetching all alerts whose device_id is in the previous list

Reply:
Alternatively, when joining, only take the columns that you need. Otherwise we keep information and use memory that doesn't need to

Reply:
The joins are done automatically on the ForeignKey columns and only the columns of the queried objects (here alerts) are retrieved.  The Database does the underlying optimization

Here is the statement for the event query (accessible simply by doing (str(<sqlalchemyQuery>) ) for example:
```
SELECT events.id AS events_id, events.lat AS events_lat, events.lon AS events_lon, events.type AS events_type, events.start_ts AS events_start_ts, events.end_ts AS events_end_ts, events.created_at AS events_created_at 
FROM events JOIN alerts ON events.id = alerts.event_id JOIN devices ON devices.id = alerts.device_id JOIN accesses ON accesses.id = devices.access_id 
WHERE accesses.group_id = %(group_id_1)s
```

Reply:
Oh damn, that's neat!